packages/paste-core/components/chat-log/__tests__/useChatLog.spec.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import {renderHook, act} from '@testing-library/react-hooks';
+
+import {useChatLog, ChatBubble, ChatMessage} from '../src';
+
+const chat = {
+  id: 'uid123',
+  content: (
+    <ChatMessage variant="inbound">
+      <ChatBubble>hi</ChatBubble>
+    </ChatMessage>
+  ),
+};
+
+describe('useChatLog', () => {
+  it('returns expected result with defaults', () => {
+    const {result} = renderHook(() => useChatLog());
+
+    expect(result.current).toMatchObject({
+      chats: [],
+      chatIds: [],
+      pop: expect.any(Function),
+      push: expect.any(Function),
+    });
+  });
+
+  it('returns expected result with initialization', () => {
+    const {result} = renderHook(() => useChatLog(chat));
+    expect(result.current.chats).toHaveLength(1);
+    expect(result.current.chatIds).toHaveLength(1);
+    expect(result.current.pop).toBeInstanceOf(Function);
+    expect(result.current.push).toBeInstanceOf(Function);
+  });
+
+  describe('push', () => {
+    it('pushes new chats without an id', () => {
+      const {result} = renderHook(() => useChatLog());
+      expect(result.current.chats).toHaveLength(0);
+      expect(result.current.chatIds).toHaveLength(0);
+
+      act(() => {
+        const chatWithoutCustomId = {...chat, id: undefined};
+        result.current.push(chatWithoutCustomId);
+      });
+
+      expect(result.current.chats).toHaveLength(1);
+      expect(result.current.chatIds).toHaveLength(1);
+    });
+
+    it('pushes new chats with an id', () => {
+      const {result} = renderHook(() => useChatLog());
+      expect(result.current.chats).toHaveLength(0);
+      expect(result.current.chatIds).toHaveLength(0);
+
+      act(() => {
+        result.current.push(chat);
+      });
+
+      expect(result.current.chats).toHaveLength(1);
+      expect(result.current.chatIds).toHaveLength(1);
+    });
+  });
+
+  describe('pop', () => {
+    it('pops chats by id', async () => {
```

Comment:
todo: can we get another pop test for without an id please? Would be fine to drop it into this `it` block and rename it to `pops chats correctly` so we don't need to create a new render